client/api/go-client/topology.go

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package client
+
+import "github.com/heketi/heketi/apps/glusterfs"
+
+func (c *Client) TopologyInfo() (*glusterfs.TopologyInfoResponse, error) {
+	var topo glusterfs.TopologyInfoResponse
+	clusterlist, err := c.ClusterList()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cluster := range clusterlist.Clusters {
+		clusteri, err := c.ClusterInfo(cluster)
+		if err != nil {
+			return nil, err
+		}
+
+		var cluster glusterfs.Cluster
+		cluster.Id = clusteri.Id
+
+		//	var volume glusterfs.VolumeInfoResponse
+		volumelist, err := c.VolumeList()
```

Comment:
This one is not necessary which caused you to add Line 50.  Instead, `clusteri` has the list of volumes already, and you can use that list in line 43.